manifests/init.pp

```diff
@@ -85,15 +87,19 @@
     '--disablesssd',
     '--disablesssdauth',
   ],
+  String $authselect_profile='sssd',
 ) {
 
   # Warn on unsupported platforms
   if ($::facts['os']['family'] == 'RedHat') {
     if ($::facts['os']['name'] == 'Amazon') and !($::facts['os']['release']['major'] in ['2']) {
       warning("osname Amazon's os.release.major is <${::facts['os']['release']['major']}> and must be 2.")
     }
-    if !($::facts['os']['name'] == 'Amazon') and !($::facts['os']['release']['major'] in ['6', '7']) {
-      warning("osfamily RedHat's os.release.major is <${::facts['os']['release']['major']}> and must be 6 or 7.")
+    if !($::facts['os']['name'] == 'RedHat') and !($::facts['os']['release']['major'] in ['6', '7', '8']) {
+      warning("osname RedHat's os.release.major is <${::facts['os']['release']['major']}> and must be 6, 7 or 8.")
+    }
+    if !($::facts['os']['name'] == 'Fedora') and !($::facts['os']['release']['major'] >= '28') {
```

Comment:
This would add support for Fedora 28 though the pull request is for RHEL8. This should be in a separate pull request. By adding it here, it confuses the two and makes both of them harder to merge. Also F28 is end of life, so we wouldn't merge it.

https://fedoraproject.org/wiki/End_of_life

Reply:
Dropped, I was trying to make the code match the doco. But I put fedora 28 instead of fedora 29.